### DISCUSSION TITLE
How can I make a fluid damage the player?

### DISCUSSION THREAD

**Discussion by thatdumbdevit** [FabricMC/fabric-api#1614] [Mod Dev Support]
Hi,
I made a liquid, but I want it to damage the player.
I tried a lot of things:

`public void onEntityCollision(BlockState state, World world, BlockPos pos, Entity entity) {
        entity.damage(DamageSource.IN_WALL, 1);
}`

![Immagine 2021-08-02 155328](https://user-images.githubusercontent.com/81831281/128186033-34bfc526-25e4-44af-b1a8-90499b00a442.png)

But nothing works.
I also tried to put change the tag of the liquid from water to water. It works, but I don't want it to do fire damage.
What can I do?

Edit:
> I also tried to put change the tag of the liquid from water to water.

Yeah, I meant water to lava, not water to water lol

**Comment by warjort** [comment]
> I also tried to put change the tag of the liquid from water to water.

I think you mean lava. There is special handling for that tag and powdered snow.

````public void onEntityCollision(boolean bl)````
I don't know what this refers to, is this something you made up?

The other one is the correct method to override on your liquid block
````public void onEntityCollision(BlockState state, World world, BlockPos pos, Entity entity)````
The logic for this is that it's invoked when the entity moves and the entity's collision box overlaps the block.
This is exactly how the damage from cactus works.

I thought your problem might be that it checks the block's collision shape (since you can walk through liquids).
But there is no such check during the onEntityCollision() processing.
In fact, bubble columns use this callback to see if they should move the player.

Without more information than "it does not work" it is hard to say what you are doing wrong.

**Reply by thatdumbdevit** [reply]
> I think you mean lava. There is special handling for that tag and powdered snow.

Yes, I meant lava. 
But anyway,

> `public void onEntityCollision(boolean bl)`
> I don't know what this refers to, is this something you made up?

Someone told me to put that because it said it was a boolean value 
![boolrn](https://user-images.githubusercontent.com/81831281/128317029-a8d5d89b-12bf-4a39-870c-d9e81b6e2e4b.PNG)

> Without more information than "it does not work" it is hard to say what you are doing wrong.

What information do you need?
